aiida/tools/importexport/dbexport/__init__.py

```diff
@@ -9,79 +9,108 @@
 from aiida.tools.importexport.dbexport.utils import (
-    check_licenses, fill_in_query, serialize_dict, check_process_nodes_sealed, summary, EXPORT_LOGGER, ExportFileFormat,
-    deprecated_parameters
+    EXPORT_LOGGER,
+    ExportFileFormat,
+    check_licenses,
+    check_process_nodes_sealed,
+    deprecated_parameters,
+    fill_in_query,
+    serialize_dict,
+    summary,
 )
-
-from .zip import ZipFolder
+from aiida.tools.importexport.dbexport.zip import ZipFolder
 
 __all__ = ('export', 'EXPORT_LOGGER', 'ExportFileFormat')
 
 
 def export(
-    entities=None,
-    filename=None,
-    file_format=ExportFileFormat.ZIP,
-    overwrite=False,
-    silent=False,
-    use_compression=True,
-    **kwargs
-):
+    entities: Optional[Iterable[Any]] = None,
+    filename: Optional[str] = None,
+    file_format: str = ExportFileFormat.ZIP,
+    overwrite: bool = False,
+    silent: bool = False,
+    use_compression: bool = True,
+    **traversal_rules: bool,
+) -> None:
     """Export AiiDA data
 
     .. deprecated:: 1.2.1
         Support for the parameters `what` and `outfile` will be removed in `v2.0.0`.
         Please use `entities` and `filename` instead, respectively.
 
-    :param entities: a list of entity instances; they can belong to different models/entities.
-    :type entities: list
+    :param entities: a list of entity instances;
+        they can belong to different models/entities.
 
-    :param filename: the filename (possibly including the absolute path) of the file on which to export.
-    :type filename: str
+    :param filename: the filename (possibly including the absolute path)
+        of the file on which to export.
 
     :param file_format: See `ExportFileFormat` for complete list of valid values (default: 'zip').
-    :type file_format: str, `ExportFileFormat`
 
-    :param overwrite: if True, overwrite the output file without asking, if it exists. If False, raise an
-        :py:class:`~aiida.tools.importexport.common.exceptions.ArchiveExportError` if the output file already exists.
-    :type overwrite: bool
+    :param overwrite: if True, overwrite the output file without asking, if it exists.
+        If False, raise an
+        :py:class:`~aiida.tools.importexport.common.exceptions.ArchiveExportError`
+        if the output file already exists.
 
     :param silent: suppress console prints and progress bar.
-    :type silent: bool
 
-    :param use_compression: Whether or not to compress the archive file (only valid for the zip file format).
-    :type use_compression: bool
+    :param use_compression: Whether or not to compress the archive file
+        (only valid for the zip file format).
 
-    :param allowed_licenses: List or function. If a list, then checks whether all licenses of Data nodes are in the
-        list. If a function, then calls function for licenses of Data nodes expecting True if license is allowed, False
+    :param allowed_licenses: List or function.
```

Comment:
if this is an allowed kwarg, then this doesn't really match with calling the kwargs `traversal_rules`
should this be made an explicit parameter?

Reply:
I'm not sure what you mean by this? the `**` syntax gathers all kwargs that are not already directly specified, so it would not include `allowed_licenses`

Reply:
Right, but `allowed_licenses` is not listed among the function arguments in the code above ;-)

Reply:
fixed in #4448